server/src/main/java/org/opensearch/gateway/AsyncShardsFetchPerNode.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.gateway;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.support.nodes.BaseNodeResponse;
+import org.opensearch.action.support.nodes.BaseNodesResponse;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.common.lease.Releasable;
+import org.opensearch.index.shard.ShardId;
+
+import java.util.Map;
+
+/**
+ * This class is responsible for fetching shard data from nodes. It is analogous to AsyncShardFetch class except
+ * that we fetch batch of shards in this class from single transport request to a node.
+ * @param <T>
+ *
+ * @opensearch.internal
+ */
+public abstract class AsyncShardsFetchPerNode<T extends BaseNodeResponse> implements Releasable {
```

Comment:
In your [PR](https://github.com/opensearch-project/OpenSearch/pull/8218) you are already created this class. This will have merge conflicts.

Please keep it in a single PR. 
Since we are going with Batch terminology, Let rename the class to `AsyncBatchShardFetch`

Reply:
I have renamed and created the exact same file at both places to avoid merge conflicts and compilation failure in CI.